source/hic_hal/stm32/stm32h743ii/extended_features/source/DAP_vendor_ex.c

```diff
@@ -283,6 +290,82 @@ uint32_t DAP_ProcessVendorCommandEx(const uint8_t *request, uint8_t *response) {
 
         break;
     }
+    case ID_DAP_VendorEx40_MEASURE_POWER:
+    {
+        bool ret = udb_power_measurement_measure();
+
+        if (ret)
+        {
+            *response = DAP_OK;
+        }
+        else
+        {
+            *response = DAP_ERROR;
+        }
+        num += 1;
+        break;
+    }
+    case ID_DAP_VendorEx41_READ_POWER:
+    {
+        uint8_t target_type = *request++;
+        uint8_t query_type = *request;
+        bool ret;
+        if (target_type < UDB_POWER_MEASUREMENT_TARGET_SIZE)
+        {
+            switch (query_type)
+            {
+                case DAP_VENDOREX_POWER_COMMAND_ARG_VOLTAGE:
+                {
+                    float voltage;
+                    ret = udb_power_measurement_read_voltage_v(target_type, &voltage);
+                    if (ret == false)
+                    {
+                        goto power_measurement_error;
+                    }
+                    *response++ = DAP_OK;
+                    *response++ = 2;
+                    uint8_t voltage_int = (int8_t)voltage;
+                    *response++ = voltage_int;
+                    // 2 digits after the decimal point
+                    *response++ = (int8_t)((voltage - voltage_int) * 100);
+                    num += 4;
+
+                    break;
+                }
+                case DAP_VENDOREX_POWER_COMMAND_ARG_CURRENT:
+                {
+                    float current;
+                    ret = udb_power_measurement_read_current_mamp(target_type, &current);
+                    if (ret == false)
+                    {
+                        goto power_measurement_error;
+                    }
+                    *response++ = DAP_OK;
+                    *response++ = 2;
+                    uint8_t current_int = (int8_t)current;
+                    *response++ = current_int;
+                    // 2 digits after the decimal point
+                    *response++ = (int8_t)((current - current_int) * 100);
```

Comment:
Is this enough precision? I'm not sure exactly what sort of DUT might be connected, but if it's possible to measure down to single microamps, that could be useful for very low-power measurements. It's not unheard of for board idle power to be that low.

Reply:
Ok, I will use microamps.